cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -183,9 +185,9 @@ public class AppLifecycleHttpHandler extends AbstractAppFabricHttpHandler {
   public BodyConsumer create(HttpRequest request, HttpResponder responder,
                              @PathParam("namespace-id") final String namespaceId,
                              @PathParam("app-id") final String appId)
-    throws BadRequestException, NamespaceNotFoundException, AccessException {
+          throws Exception {
 
-    ApplicationId applicationId = validateApplicationId(namespaceId, appId);
+    ApplicationId applicationId = validateApplicationVersionId(namespaceId, appId, UUID.randomUUID().toString());
 
```

Comment:
A new version is created on deploy action (instead of defaulting to -SNAPSHOT.
The version-id is a UUID

Reply:
Can we use a time-based UUID instead of random one? You can use the `RunIds.generate()` for that.

Reply:
changes it to `RunIds.generate().getId()`